### PR TITLE
Clean concepts - models docs

### PR DESCRIPTION
**Pull request by robertpiosik**
#### Description of what you did:

- remove capitalisation from file paths as models create lowercased files, eg. `restaurant.settings.json`, `restaurant.js`
- add `async` to code snippet in lifecycle callbacks
- rename middle parameter in `afterSave` hook
- add information how many hooks are there under user's disposal (10)
- to avoid confusion, model definition example should contain informations that are generated by default - it should include `options` property
- model definition file - `info.name` should be capitalised 
- simplify `attributes.name`, `attributes.description` by removing empty `default` fields. Change order of attributes to start with simpler types and end with relational
- component is, similarly to model - a database structure. Avoid confusion and stick with the same vocabulary. Change slightly description of a component model.
- add real world example to component model description
- remove dot from relative path in section "Where are the models defined?" making it an absolute path to project root
- move command in `For Content Types models` to the new line
- remove typo in word `bellow`
- modify example of a model settings to better show the purpose of custom definition (version 2 of the model)
- add description to `info` key descriprion
- rename User to restaurant in `Model options` to be consistent with other examples on the page

 ### Concern:
[Bookshelf example](https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#example-9) states that each callback receives `attrs` but in the actual autogenerated file it varies between `attrs` `response` and `columns`. Shouldn't we be more strict here?

### CODE REVIEW

File: docs/3.0.0-beta.x/concepts/models.md [hunk 1]
```diff
@@ -91,12 +95,16 @@ If you are just starting out it is very convenient to generate some models with
 
 ### For Content Types models
 
-Use the CLI, and run the following command `strapi generate:model restaurant name:string description:text`.<br>Read the [CLI documentation](../cli/CLI.md) for more information.
+Use the CLI, and run the following command:
 
-This will create two files located at `./api/restaurant/models`:
+`strapi generate:model restaurant name:string description:text`
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/concepts/models.md:100]: Can you please use the tab system to have the `yarn`, `npx` and `strapi` commands

Like here - http://localhost:8080/documentation/3.0.0-beta.x/installation/cli.html#step-2-create-a-new-project

File: docs/3.0.0-beta.x/concepts/models.md [hunk 2]
```diff
@@ -106,31 +114,34 @@ When you create a new API using the CLI (`strapi generate:api <name>`), a model
...
   "connection": "mongo",
-  "collectionName": "Restaurants_v1",
-  "globalId": "Restaurants",
+  "globalId": "restaurantsV2",
   "attributes": {}
 }
 ```
 
-In this example, the model `Restaurant` will be accessible through the `Restaurants` global variable. The data will be stored in the `Restaurants_v1` collection or table and the model will use the `mongo` connection defined in `./config/environments/**/database.json`
+In this example, the model `Restaurants V2` will be accessible through the `restaurantsV2` global variable. The data will be stored in the `restaurants_v2` collection (mongoDB) or table (SQL) and the model will use the `mongo` connection defined in `./config/environments/**/database.json`.
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/concepts/models.md:144]:
```diff
- mongoDB
+ MongoDB
```

Comment (robertpiosik) [reply docs/3.0.0-beta.x/concepts/models.md]: done

File: docs/3.0.0-beta.x/concepts/models.md (hunk 1 above)
Comment (robertpiosik) [reply docs/3.0.0-beta.x/concepts/models.md]: done

**Comment by robertpiosik** [comment]
> Thank you for these changes, I really appreciate.
> 
> Another comment for every lists.
> You changed `:` by `-` or `—`
> Can you use `:` for every list please, to keep a consistency with other pages.
> 
> Thanks

all done

**Comment by robertpiosik** [comment]
@lauriejim what's the status of this PR? Do we have any blocker?

**Comment by alexandrebodin** [comment]
Hi @robertpiosik We released new doc for the v3.x Would you be willing to make your PR in those docs instead ? Thanks !

**Comment by robertpiosik** [comment]
@alexandrebodin on it!:)

**Pull request closed without merging**